raiden-dapp/src/service-worker/assistant.ts

```diff
@@ -2,75 +2,161 @@
-  };
+  }
+
+  private onMessage(event: MessageEvent): void {
+    if (!(event.data && event.data.messageIdentifier)) return;
 
-  private onMessage = (event: MessageEvent): void => {
-    if (!event.data) return;
+    const { messageIdentifier, ...payload } = event.data;
 
-    switch (event.data) {
-      case ServiceWorkerMessages.INSTALLATION_ERROR:
-      case ServiceWorkerMessages.CACHE_IS_INVALID:
+    switch (messageIdentifier) {
+      case ServiceWorkerMessageIdentifier.INSTALLED_VERSION:
+        this.setInstalledVersion(payload.version);
+        break;
+
+      case ServiceWorkerMessageIdentifier.INSTALLATION_ERROR:
+        this.reportInstallationError(payload.error);
+        break;
+
+      case ServiceWorkerMessageIdentifier.CACHE_IS_INVALID:
         this.setUpdateIsMandatory();
         break;
 
-      case ServiceWorkerMessages.RELOAD_WINDOW:
+      case ServiceWorkerMessageIdentifier.RELOAD_WINDOW:
         this.reloadWindow();
         break;
 
       default:
         break;
     }
-  };
+  }
 
-  private updateAvailableVersion = async (): Promise<void> => {
+  private reportInstallationError(error: Error): void {
+    this.console.error('Service worker failed during installation phase.');
+    this.console.error(error);
+    this.setUpdateIsMandatory();
+  }
+
+  private async updateAvailableVersion(): Promise<void> {
     try {
-      const response = await fetch(VERSION_FILE_PATH);
+      const response = await this.window.fetch(VERSION_FILE_PATH);
       const data = await response.json();
       const version = data.version.version;
-
-      if (validateVersion(version)) {
-        this.store.commit('setAvailableVersion', version);
-      } else {
-        throw new Error(`Maleformed version string: ${version}`);
-      }
+      this.setAvailableVersion(version);
     } catch (error) {
-      console.warn(`Failed to get (a valid) version: ${(error as Error).message}`); // eslint-disable-line no-console
+      this.console.error('Failed to update available version');
+      this.console.error(error);
+    }
+  }
+
+  private verifyCacheValidity(): void {
+    this.postMessage(ServiceWorkerAssistantMessageIdentifier.VERIFY_CACHE);
+  }
+
+  async isAnyServiceWorkerRegistered(): Promise<boolean> {
+    const allServiceWokers = (await this.serviceWorkerContainer?.getRegistrations()) ?? [];
+    return allServiceWokers.length > 0;
+  }
+
+  private async postMessage(
+    messageIdentifier: ServiceWorkerAssistantMessageIdentifier,
+  ): Promise<void> {
+    if (this.serviceWorkerContainer) {
+      await this.serviceWorkerContainer.ready;
+      const message = { messageIdentifier };
+      this.serviceWorkerContainer.controller?.postMessage(message);
+    }
+  }
+
+  private async unregisterAllServiceWorker(): Promise<void> {
+    const allServiceWokers = (await this.serviceWorkerContainer?.getRegistrations()) ?? [];
+
+    for (const serviceWorker of allServiceWokers) {
+      await serviceWorker.unregister();
     }
```

Comment:
I was thinking about something like this but was too lazy. 😆 
But because it is about performance here so the user has not much lagging it is important! 👍🏾

Reply:
✔️